### PR TITLE
[Dubbo-3707] Merge dubbo-serialization-gson into incubator-dubbo

### PR DESCRIPTION
**Pull request by tswstarplanet**
## What is the purpose of the change

#3707 

## Brief changelog

XXXXX

## Verifying this change

XXXXX

Follow this checklist to help us incorporate your contribution quickly and easily:

- [x] Make sure there is a [GITHUB_issue](https://github.com/apache/incubator-dubbo/issues) field for the change (usually before you start working on it). Trivial changes like typos do not require a GITHUB issue. Your pull request should address just this issue, without pulling in other changes - one PR resolves one issue.
- [ ] Format the pull request title like `[Dubbo-XXX] Fix UnknownException when host config not exist #XXX`. Each commit in the pull request should have a meaningful subject line and body.
- [ ] Write a pull request description that is detailed enough to understand what the pull request does, how, and why.
- [ ] Write necessary unit-test to verify your logic correction, more mock a little better when cross module dependency exist. If the new feature or significant change is committed, please remember to add integration-test in [test module](https://github.com/apache/incubator-dubbo/tree/master/dubbo-test).
- [ ] Run `mvn clean install -DskipTests=false` & `mvn clean test-compile failsafe:integration-test` to make sure unit-test and integration-test pass.
- [ ] If this contribution is large, please follow the [Software Donation Guide](https://github.com/apache/incubator-dubbo/wiki/Software-donation-guide).

### CODE REVIEW

**Comment by tswstarplanet** [comment]
I have run the unit test, and write a demo to test the new module

File: dubbo-serialization/dubbo-serialization-gson/pom.xml [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  See the License for the specific language governing permissions and
+  limitations under the License.
+  -->
+<project xmlns="http://maven.apache.org/POM/4.0.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
+    <modelVersion>4.0.0</modelVersion>
+    <parent>
+        <artifactId>dubbo-serialization</artifactId>
+        <groupId>org.apache.dubbo</groupId>
+        <version>2.7.1-SNAPSHOT</version>
```
Comment (CrazyHZM) [inline dubbo-serialization/dubbo-serialization-gson/pom.xml]: Hi, since 2.7.1 has been released, could you update it to 2.7.2-SNAPSHOT?

Comment (tswstarplanet) [reply dubbo-serialization/dubbo-serialization-gson/pom.xml]: OK, I will do it tonight !

File: dubbo-common/pom.xml [hunk 2]
```diff
@@ -77,5 +77,9 @@
             <groupId>de.ruedigermoeller</groupId>
             <artifactId>fst</artifactId>
         </dependency>
+        <dependency>
```
Comment (ralf0131) [inline dubbo-common/pom.xml]: It is unnecessary to add the dependency here.

File: dubbo-serialization/dubbo-serialization-api/pom.xml [hunk 3]
```diff
@@ -55,5 +55,9 @@ limitations under the License.
             <groupId>de.ruedigermoeller</groupId>
             <artifactId>fst</artifactId>
         </dependency>
+        <dependency>
+            <groupId>com.google.code.gson</groupId>
```
Comment (ralf0131) [inline dubbo-serialization/dubbo-serialization-api/pom.xml]: What is the dependencies of gson? Are all the licenses compatible with Apache?

**Comment by ralf0131** [comment]
Sorry, there is some conflict, I think it is due to the avro serialization has been merged. Could you please resolve the conflict?

File: dubbo-serialization/dubbo-serialization-api/pom.xml (hunk 3 above)
Comment (tswstarplanet) [reply dubbo-serialization/dubbo-serialization-api/pom.xml]: This package is copy from the project dubbo/dubbo-serialization-gson. I think it's OK ?

**Comment by tswstarplanet** [comment]
> Sorry, there is some conflict, I think it is due to the avro serialization has been merged. Could you please resolve the conflict?

OK, I will resolve the conflict.

File: dubbo-serialization/dubbo-serialization-api/pom.xml (hunk 3 above)
Comment (ralf0131) [reply dubbo-serialization/dubbo-serialization-api/pom.xml]: It has not been checked before. When merging into Apache repo, we need to be careful to the dependencies.
Could you list all the dependencies of gson and there license?

Comment (ralf0131) [reply dubbo-serialization/dubbo-serialization-api/pom.xml]: Hi, I just checked that the gson library has already been added before this pull request. So I think there is no issue here.

**Pull request merged**